### DISCUSSION TITLE
How to set a lower bound for a dimension?

### DISCUSSION THREAD

**Discussion by lewis-xuwei** [google/or-tools#3380] [Routing (and legacy CP) questions] [answered]
```
dist_dimension = "dist"
self.routing_solver.AddDimension(
            dist_callback_index,
            0,  
            self.routeTotalLengthUB,
            True,
            dist_dimension,
        )
```
For instance, I can set the upper bound for dist. however, how to set a lower bound for a dimension is still in the air.

**Comment by hklarnerben** [comment, marked as answer]
Not sure if this is a good idea, but you could set a constraint on the cumulative variable at the end of each vehicle:

```
index = routing.End(vehicle_id)
dimension.CumulVar(index).SetMin(value)
```

**Reply by lewis-xuwei** [reply]
Thanks.

**Reply by lewis-xuwei** [reply]
Plus, do you know how to set the profit of a pair of pickup and delivery?

**Reply by hklarnerben** [reply]
Hm. Maybe you can find a way of lowering the cost transit callback for transitions to the pickup and delivery?
